plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -194,6 +221,60 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 Optional.ofNullable(nameMappingJson));
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
```

Comment:
It's quite difficult to test `DATE` type, I'm still thinking about how to simulate that... meanwhile I am just supporting the 2 timestamp types.

Reply:
It's also unclear to me if we should take the beginning of a day or end of a day for time travel, so I think it's better to just not support it until explicit user request.

Reply:
I agree, using dates would be problematic as Iceberg stores the snapshot timestamp in millis.